velox/functions/sparksql/In.cpp

```diff
@@ -33,14 +32,14 @@ class Set<StringView> {
   using value_type = std::string_view;
 
   void emplace(const StringView& s) {
-    std::string_view sv(s.data(), s.size());
+    std::string sv(s.data(), s.size());
```

Comment:
This will regress peformance.

Using an Arena improves cache locality and reduces memory management overhead. Using a std::string undermines this. There's a benchmark for `In` that you can use to confirm this.

If the primary issue with Arena was simply that it does not track memory, can we not just pass in a `MemoryPool`? There really should be useable Arena implementation in Velox.

Reply:
Also `Set<StringView>` is a very misleading performance hint if we keep actual strings inside.

Reply:
@funrollloops This particular Arena implementation is very primitive. Proper Arena implementation is in velox/common/memory/HashStringAllocator.h . To unblock this change, we could move Arena.h to velox/functions/sparksql and keep In.cpp unmodified.

Reply:
Just move `Arena.h` into `velox/functions/sparksql` for now. `HashStringAllocator` is too complicated for me to quickly understand, but I chose this `Arena` implementation because it did exactly what I wanted.

Reply:
@funrollloops Sagar, I took a closer look at how Set<StringView> is used and I think you don't need to copy the strings at all. Instead you can simply hold on to string buffers from the `elementsVector` and store StringViews directly (without copying) in the set.

Reply:
I'll investigate if I have some time.

If the input strings are not already close together in memory, it's probably worth copying them. Also it's possible using `StringView` is faster than `std::string_view`, I can't recall if I explicitly tested that. `std::string_view` avoids an extra branch on the string size, but `StringView` would be more compact (due to inlined strings); the latter is probably better to allow for a larger number of strings to fit into cache.